kaiax/staking/impl/getter.go

```diff
@@ -123,11 +123,14 @@ func (s *StakingModule) getFromState(header *types.Header, statedb *state.StateD
 		return nil, staking.ErrAddressBookCall(err)
 	}
 
-	// Get CL registry info if Prague fork is enabled
+	// Get CL registry info if staking info is for Prague block
 	var clRes clRegistryResult
-	if isPrague {
+	if isForPrague {
 		// If Registry is not installed, do not handle CL staking info.
-		if statedb.GetCode(system.RegistryAddr) == nil {
+		// In private network, Randao and Prague hardfork can be activated at the same block.
+		// It leads to staking info inconsistency between block processing and rpc query since the Registry hasn't been installed when finalizing the header.
+		// Note that Randao can't be activated after Prague according to fork ordering (Randao <= Kaia <= Prague).
+		if statedb.GetCode(system.RegistryAddr) == nil || s.ChainConfig.IsRandaoForkBlockParent(header.Number) {
```

Comment:
Decription considers staking info inconsistency by absent of registry contract. Should we check the Randaok fork for all checks of registry installation, or can we skip checking it because it's definite that the registry contract exists after Randao(i.e., unnecessary check)?

Reply:
We need to check the Randao fork for registry installation if it leads to any inconsistencies in block processing.

For example, we don't need to check this for APIs like `Kaia.getActiveAddressFromRegistry`. When passing the `Randao` block, it just acts like it has a Registry even if there was no Registry before `InstallRegistration` in block processing.

Reply:
Given inconsistency,
inconsistency = staking amount calculation (+CL)

Then, I guess L134 should be triggered if registry is nil and before prague. My question was, why is it Randao check rather Prague?

Reply:
This can only happen in the private networks where `randaoCompatibleBlock == pragueCompatibleBlock.`

Reply:
Found it, thank you.